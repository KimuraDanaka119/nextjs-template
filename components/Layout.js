import React from 'react';
import Router from 'next/router';
import NProgress from 'nprogress';

import Header from './Header';
import Footer from './Footer';

// progress bar
Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => NProgress.done();
Router.onRouteChangeError = () => NProgress.done();

const Layout = props => [
  <Header key="0" imagePath={props.config.paths.images}/>,
  <div key="1" className="container">
    {props.children}
  </div>,
  <Footer key="2" />,
];

export default Layout;
