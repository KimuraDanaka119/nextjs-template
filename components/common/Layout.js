import React from 'react';
import PropTypes from 'prop-types';
import Router from 'next/router';
import NProgress from 'nprogress';

import Header from './Header';
import Footer from './Footer';

// progress bar
Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => NProgress.done();
Router.onRouteChangeError = () => NProgress.done();

const Layout = props => (
  <div>
    <Header />
    <div className="container">
      {props.children}
    </div>
    <Footer />
  </div>
);

Layout.propTypes = {
  children: PropTypes.element,
};

export default Layout;
