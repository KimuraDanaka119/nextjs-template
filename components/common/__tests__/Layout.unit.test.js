import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Layout from '../Layout';

describe('<Layout />', () => {
  it('should render correctly with no children', () => {
    const layout = shallow(<Layout/>);
    expect(toJson(layout)).toMatchSnapshot();
  });

  it('should render correctly with some children', () => {
    const layout = shallow(<Layout>
      <div>Hello World 1</div>
      <p>Hello World 2</p>
      <span>Hello World 3</span>
    </Layout>);
    expect(toJson(layout)).toMatchSnapshot();
  });
});
