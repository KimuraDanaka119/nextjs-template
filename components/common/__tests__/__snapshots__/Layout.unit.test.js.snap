// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render correctly with no children 1`] = `
<div>
  <Translate(I18NComponent) />
  <div
    className="container"
  />
  <Translate(I18NComponent) />
</div>
`;

exports[`<Layout /> should render correctly with some children 1`] = `
<div>
  <Translate(I18NComponent) />
  <div
    className="container"
  >
    <div>
      Hello World 1
    </div>
    <p>
      Hello World 2
    </p>
    <span>
      Hello World 3
    </span>
  </div>
  <Translate(I18NComponent) />
</div>
`;
